sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -3406,42 +3406,1131 @@ lsc_format_ret(__ESIMD_NS::simd<T1, N> Vals) {
+  constexpr uint16_t AddressScale = 1;
+  constexpr int ImmOffset = 0;
+  constexpr lsc_data_size EDS = expand_data_size(finalize_data_size<T, DS>());
+  constexpr lsc_vector_size VS = to_lsc_vector_size<1>();
+  constexpr lsc_data_order Transposed = lsc_data_order::nontranspose;
+  using MsgT = typename lsc_expand_type<T>::type;
+  constexpr int IOp = lsc_to_internal_atomic_op<T, Op>();
+  simd<MsgT, N> Msg_data0 = lsc_format_input<MsgT>(src0);
+  simd<MsgT, N> Msg_data1 = lsc_format_input<MsgT>(src1);
+  simd<MsgT, N> Tmp =
+      __esimd_lsc_xatomic_slm_2<MsgT, IOp, cache_hint::none, cache_hint::none,
+                                AddressScale, ImmOffset, EDS, VS, Transposed,
+                                N>(pred.data(), offsets.data(),
+                                   Msg_data0.data(), Msg_data1.data());
+  return lsc_format_ret<T>(Tmp);
+}
+
 } // namespace detail
 
-/// Atomic update operation performed on SLM. No source operands version.
-/// See description of template and function parameters in @ref
-/// usm_atomic_update0 "atomic update" operation docs.
-template <atomic_op Op, typename Tx, int N, class T = detail::__raw_t<Tx>>
-__ESIMD_API simd<Tx, N> slm_atomic_update(simd<uint32_t, N> offsets,
-                                          simd_mask<N> mask) {
-  detail::check_atomic<Op, T, N, 0>();
-  const auto si = __ESIMD_NS::get_surface_index(detail::LocalAccessorMarker());
-  return __esimd_dword_atomic0<Op, T, N>(mask.data(), si, offsets.data());
+/// @anchor slm_atomic_update0
+/// @brief Atomic update operation performed on SLM.
+/// No-argument variant of the atomic update operation.
+
+/// simd<T, N>
+/// slm_atomic_update(simd<uint32_t, N> byte_offset,
+///                   simd_mask<N> mask = 1);                   /// (slm-au0-1)
+///
+/// simd<T, N>
+/// slm_atomic_update(simd_view<simd<uint32_t, N>, RegionTy> byte_offset,
+///                   simd_mask<N> mask = 1);                   /// (slm-au0-2)
+///
+
+/// The following functions do the same work as slm_atomic_update(). They accept
+/// a local accessor \p lacc and the atomic update is done from SLM associated
+/// with \p lacc plus \p byte_offset applied to it. If \p byte_offset
+/// is omitted, then zero offset is used.
+
+/// simd<T, N> atomic_update(local_accessor lacc,
+///                          simd<uint32_t, N> byte_offset,
+///                          simd_mask<1> pred = 1);
+///                                                             // (lacc-au0-1)
+/// simd<T, N> atomic_update(local_accessor lacc,
+///                          simd_view<simd<uint32_t, N>, RegionTy> byte_offset,
+///                          simd_mask<1> pred = 1);
+///                                                             // (lacc-au0-2)
+/// simd<T, N> atomic_update(local_accessor lacc,
+///                          simd_mask<1> pred = 1);
+///                                                             // (lacc-au0-3)
```

Comment:
We agreed not to provide API with scalar offset for atomic_update and gather/scatter.
Please see at (acc-au-1,2,3,4) variants.
I believe this (lacc-au0-3) should be removed.

Reply:
Done.